checks/network.py

```diff
@@ -57,3 +57,19 @@ def checkNICSpeed(lines):
                     if 'GbE' in nic or 'Gigabit' in nic:
                         return [LEVEL_WARNING, "Slow Network Connection", "Your gigabit-capable network card is only connecting at 100mbps. This may indicate a bad network cable or outdated router / switch which could be impacting network performance."]
     return None
+
+
+x264stream_re = re.compile(r"stream")
+
+
+def checkDynamicBr(lines):
+    dynBrLines = search('Dynamic bitrate enabled', lines)
+    if (len(dynBrLines) > 0):
+        x264Lines = search('x264 encoder: ', lines)
+        for i in x264Lines:
+            if x264stream_re.search(i):
+                return [LEVEL_INFO, "Dynamic Bitrate", "Dynamic Bitrate is enabled. This does not resolve network issues, and will reduce bitrate (reduce quality) instead of dropping frames."]
+        else:
+            return [LEVEL_WARNING, "Dynamic Bitrate",
+                    """Dynamic bitrate is enabled and a hardware encoder is potentially in use. This can cause issues with hardware encoders if bitrate changes happen too frequently, or drops too low. Should you experience bitrate dropping to zero, or no output even if OBS says its streaming, either switch to x264 or disable dynamic bitrate thru settings->Advanced, uncheck 'dynamically change bitrate'"""]
```

Comment:
> [..] either switch to x264 or disable dynamic bitrate thru settings->Advanced, uncheck 'dynamically change bitrate'

Please capitalise Settings, surround breadcrumb arrows with spaces, and end sentences with a dot. Personally, when talking about Settings I prefer to use "in" rather than "through."

Here's how I'd write it:

> [..] either change your Encoder to x264 or turn off Dynamic Bitrate in Settings -> Advanced -> Network.